### PR TITLE
#604 PsAll.java:56-60: PsAll ctor. 

### PR DESCRIPTION
**Pull request by xupyprmv**
#604 Parameter validation was moved to separate private method `validate`.

### CODE REVIEW

**Comment by davvd** [comment]
@xupyprmv Thanks, I will find someone to review this PR soon

**Comment by davvd** [comment]
@carlosmiranda please review,thanks

**Comment by carlosmiranda** [comment]
@rultor merge

File: src/main/java/org/takes/facets/auth/PsAll.java [hunk 1]
```diff
@@ -100,6 +82,28 @@ public final Response exit(final Response response, final Identity identity)
...
 
     /**
+     * Validate index.
+     * @param idx Index of a Pass whose Identity to return on successful
+     *  {@link PsAll#enter(Request)}
+     * @return Validated index
+     */
+    private int validated(final int idx) {
+        if (idx < 0) {
+            throw new IllegalArgumentException("Index must be >= 0");
```
Comment (yegor256) [inline src/main/java/org/takes/facets/auth/PsAll.java]: @xupyprmv would be nice to show the value provided, see http://www.yegor256.com/2015/12/01/rethrow-exceptions.html

File: src/main/java/org/takes/facets/auth/PsAll.java [hunk 2]
```diff
@@ -100,6 +82,28 @@ public final Response exit(final Response response, final Identity identity)
...
+        }
+        if (idx >= this.all.size()) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Trying to return index %s from a list of %s passes",
```
Comment (yegor256) [inline src/main/java/org/takes/facets/auth/PsAll.java]: @xupyprmv it's better to use `%d` when printing numbers

**Comment by yegor256** [comment]
@xupyprmv a few comments above. any tests for that?

**Comment by xupyprmv** [comment]
@yegor256 Thanks for CR. I modified code according your remarks and added one more test. Please continue CR.

**Comment by xupyprmv** [comment]
@yegor256 please continue with CR

**Comment by xupyprmv** [comment]
@yegor256 please continue with CR

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by ypshenychka** [comment]
@carlosmiranda 
According to our [QA Rules](http://at.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.

No issues were found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by ypshenychka** [comment]
@xupyprmv 
>#604 PsAll.java:56-60: PsAll ctor.

According to our [QA Rules](http://at.teamed.io/qa.html):
>Pull request title explains the problem it is fixing.

Please update title to be more informative. And puzzle name "PsAll.java:56-60" is redundant for PRs title.

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda Thanks.

**Comment by xupyprmv** [comment]
@ypshenychka Title was updated.

**Comment by ypshenychka** [comment]
@xupyprmv Thank you.

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the review, we'll try better next time

**Comment by davvd** [comment]
@carlosmiranda paied 10 mins to @ypshenychka for QA review (payment ID is `79206155`); Thanks for your contribution, **17 mins** was added to your account, payment ID is `79206179`, 336 hours and 14 mins spent total; extra minutes for review comments (c=2); +17 to your rating, your total score is [+3176](http://www.netbout.com/b/35023?open=rating)

**Pull request merged**